lib/IlmCtl/CtlRcPtr.cpp

```diff
@@ -96,7 +97,7 @@ throwRcPtrExc (const RcObject *lhs, const RcObject *rhs)
 Mutex &
 rcPtrMutex (RcObject *ptr)
 {
-    return mutexes[(unsigned long)ptr % NUM_MUTEXES];
+    return mutexes[reinterpret_cast<intptr_t>(ptr) % NUM_MUTEXES];
```

Comment:
Does this creates any compilation issues, if so in which platform, shall I know the motivation for this change?

Just curious to know the reason for this change :)